sdk/src/integrationTest/java/TokenUpdateIntegrationTest.java

```diff
@@ -684,4 +688,1812 @@ void cannotUpdateNonFungibleTokenMetadataWhenMetadataKeyNotSet() throws Exceptio
 
         testEnv.close(tokenId);
     }
+
+    /**
+     * @notice E2E-HIP-540
+     * @url https://hips.hedera.com/hip/hip-540
+     */
+    @Test
+    @DisplayName("Can make a token immutable when updating keys to an empty KeyList, signing with an Admin Key, and setting the key verification mode to NO_VALIDATION")
+    void canMakeTokenImmutableWhenUpdatingKeysToEmptyKeyListSigningWithAdminKeyWithKeyVerificationSetToNoValidation() throws Exception {
```

Comment:
Integration tests are usually grouped by functionality and moving these tests to another file would break the group as HIP-540 is part of TokenUpdate. Also there are comments above every integration test saying this is a test for HIP-540 which in my opinion is enough for separating with the other TokenUpdate integration tests. The only disadvantage is the file is pretty big, but keeping these tests at the bottom lines of the file should be good to go.

Reply:
@0xivanov I am for keeping it all in one file